services/src/main/java/org/fao/geonet/api/records/MetadataApi.java

```diff
@@ -404,6 +404,10 @@ private Object getRecordAs(
             }
         }
 
+        if (!metadataUtils.isMetadataAvailableInPortal(Integer.parseInt(mdId))) {
+            throw new NotAllowedException(ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW);
```

Comment:
How is the portal supposed to work? If it is supposed to hide the data then in this case it should return a 404 as if it was not found in that portal.  A not allowed exception implies that the record does exists in the portal but the user is not allowed to see it. This can be a little confusing if the user is an administrator and trying to figure out why an admin is now allowed to view a record? In either case - not sure if there should be a DEBUG log entry indicating that data is not allowed due to record not belonging to that portal?

Reply:
@ianwallen I've updated the pull request: https://github.com/geonetwork/core-geonetwork/pull/8263/commits/ea85dc83f2d3ee89fa518f0680914c9f008b793e